lib/pbench/agent/tool_data_sink.py

```diff
@@ -16,38 +14,61 @@
 import json
 import logging
 import os
+import shutil
 import subprocess
 import sys
 import tempfile
+import time
 
+from configparser import ConfigParser, DuplicateSectionError
+from datetime import datetime
 from distutils.spawn import find_executable
 from http import HTTPStatus
+from jinja2 import Environment, FileSystemLoader
 from pathlib import Path
 from threading import Thread, Lock, Condition
 from wsgiref.simple_server import WSGIRequestHandler, make_server
-from jinja2 import Environment, FileSystemLoader
 
 import daemon
 import pidfile
 import redis
 
 from bottle import Bottle, ServerAdapter, request, abort
 
-import pbench.agent.toolmetadata as toolmetadata
 from pbench.agent import PbenchAgentConfig
+from pbench.agent.constants import (
+    tds_port,
+    tm_allowed_actions,
+    tm_channel_suffix_from_client,
+    tm_channel_suffix_from_tms,
+    tm_channel_suffix_to_client,
+    tm_channel_suffix_to_logging,
+    tm_channel_suffix_to_tms,
+)
+from pbench.agent.redis import RedisChannelSubscriber
+from pbench.agent.toolmetadata import ToolMetadata
+from pbench.agent.utils import collect_local_info
 
 
 # Read in 64 KB chunks off the wire for HTTP PUT requests.
 _BUFFER_SIZE = 65536
 
+# Maximum size of the tar ball for collected tool data.
+_MAX_TOOL_DATA_SIZE = 2 ** 30
+
 # Executable path of the tar program.
 tar_path = None
+cp_path = None
 
-# FIXME: move to a constants area, or a configuration setting.
-client_channel = "tool-meister-client"
 
-# FIXME: move to a constants area, or a configuration setting.
-_MAX_TOOL_DATA_SIZE = 2 ** 30
+def _now(when):
+    """_now - oh this is ugly
```

Comment:
Arguably true, and yet rather unhelpful. It's like the Gosmacs tty driver module where the only comment was "You don't want to know"...